### PR TITLE
Bug 1884131: Use lib-go buildmachinery

### PR DESCRIPTION
**Pull request by stlaz**
First part of a more complex move to the buildmachinery and aws-e2e tests.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,14 +1,27 @@
...
+IMAGE_REGISTRY?=registry.svc.ci.openshift.org
+
+# This will call a macro called "build-image" which will generate image specific targets based on the parameters:
+# $0 - macro name
+# $1 - target name
+# $2 - image ref
+# $3 - Dockerfile path
+# $4 - context directory for image build
+# It will generate target "image-$(1)" for building the image and binding it as a prerequisite to target "images".
+$(call build-image,ocp-oauth-proxy,$(IMAGE_REGISTRY)/ocp/4.4:oauth-proxy,./Dockerfile.rhel7,.)
```
Comment (tnozicka) [inline Makefile:19]: 4.6

File: go.mod [hunk 2]
```diff
@@ -9,6 +9,7 @@ require (
 	github.com/bmizerany/assert v0.0.0-20160611221934-b7ed37b82869
 	github.com/fsnotify/fsnotify v1.4.9
 	github.com/mreiferson/go-options v1.0.0
+	github.com/openshift/build-machinery-go v0.0.0-20200819073603-48aa266c95f7
```
Comment (tnozicka) [inline go.mod:12]: nit: this technically goes to bump commit

File: string_array_test.go [hunk 3]
```diff
@@ -1,9 +1,8 @@
 package main
 
 import (
-	"testing"
-
 	"github.com/bmizerany/assert"
+	"testing"
```
Comment (tnozicka) [inline string_array_test.go:5]: we usually group std lib separately

File: util/util_test.go [hunk 4]
```diff
@@ -70,7 +70,10 @@ func TestGetCertPool(t *testing.T) {
 	}
 	assert.Equal(t, "Invalid empty list of Root CAs file paths", err.Error())
 
-	tempDir := os.TempDir()
+	tempDir, err := ioutil.TempDir(os.TempDir(), "oauth-proxy-test")
```
Comment (tnozicka) [inline util/util_test.go:73]: `t.TempDir()`?

File: Makefile [hunk 5]
```diff
@@ -1,14 +1,27 @@
...
 
 clean:
 	$(RM) ./oauth-proxy
 .PHONY: clean
+
+GO_BUILD_PACKAGES := .
+# avoid the test/ directory only containing the e2e tests
+GO_TEST_PACKAGES :=./ ./api/... ./cookie/... ./providers/... ./util/...
```
Comment (tnozicka) [inline Makefile:27]: Where is the target to run local e2es?

Comment (stlaz) [reply Makefile:27]: that's tbd in #190, currently no tests are being run. Well, they are, in Travis, they attempt to build origin 3.11 and fail horribly. This repo is a mess.

**Comment by tnozicka** [comment]
also drop travis

File: go.mod (hunk 2 above)
Comment (stlaz) [reply go.mod:12]: I'd keep it here as "adding a new dependency due to changes in the repo", I'm keeping the vendoring in a separate commit though

File: Makefile (hunk 5 above)
Comment (tnozicka) [reply Makefile:27]: I suppose I'd at least hook it as when setting up the makefile:
```
test-e2e: GO_TEST_PACKAGES :=./test/e2e/...
test-e2e: GO_TEST_FLAGS += -v
test-e2e: GO_TEST_FLAGS += -timeout 3h
test-e2e: GO_TEST_FLAGS += -p 1
test-e2e: test-unit
.PHONY: test-e2e
``

File: go.mod (hunk 2 above)
Comment (tnozicka) [reply go.mod:12]: ok, we don't enforce it yet, but be prepared for "every change to go.{mod,sum} or vendor/" must start with "bump:" message

File: Makefile (hunk 5 above)
Comment (stlaz) [reply Makefile:27]: sure, I have that in a successive PR #148, but I'll add it here, too

Comment (tnozicka) [reply Makefile:27]: ok, you linked this PR before so I wasn't sure

**Pull request merged**